benchmarks/devel/devel.cpp

```diff
@@ -0,0 +1,64 @@
+/****************************************************************************
+ * Copyright (c) 2017-2022 by the ArborX authors                            *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the ArborX library. ArborX is                       *
+ * distributed under a BSD 3-clause license. For the licensing terms see    *
+ * the LICENSE file in the top-level directory.                             *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+#include <Kokkos_Core.hpp>
+
+#include <chrono>
+
+#include <benchmark/benchmark.h>
+
+template <typename ExecutionSpace>
+void BM_benchmark(benchmark::State &state)
+{
+  ExecutionSpace exec_space;
+
+  auto const n = state.range(0);
+
+  Kokkos::View<int *, typename ExecutionSpace::memory_space> view(
+      Kokkos::view_alloc(exec_space, "Benchmark::view",
+                         Kokkos::WithoutInitializing),
+      n);
+
+  for (auto _ : state)
+  {
+    exec_space.fence();
+    auto const start = std::chrono::high_resolution_clock::now();
+
+    Kokkos::parallel_for(
+        "Benchmark::iota",
+        Kokkos::RangePolicy<ExecutionSpace>(exec_space, 0, n),
+        KOKKOS_LAMBDA(int i) { view(i) = i; });
+
+    exec_space.fence();
+    auto const end = std::chrono::high_resolution_clock::now();
+    std::chrono::duration<double> elapsed_seconds = end - start;
+    state.SetIterationTime(elapsed_seconds.count());
+  }
+  state.counters["rate"] =
```

Comment:
If we really put a user defined counter (which I am not convinced we should) I would prefer us doing bandwidth (times `sizeof(bla)`)

Reply:
Disagree. Rate is independent of the kernel. Bandwidth is not. Extra stuff to remember.

Reply:
Well if we don't set a counter there is no argument any more